aubio-rs/src/vec.rs

```diff
@@ -324,3 +325,173 @@ impl<'a, T: AsMut<[f32]>> From<T> for CVecPhasMut<'a> {
         }
     }
 }
+
+/**
+ * Immutable matrix of real valued data.
+ */
+#[repr(C)]
+pub struct FMat<'a, X> {
+    fmat: ffi::fmat_t,
+    _x: X,
+    _pd: PhantomData<&'a ()>,
+}
+
+impl<'a, X> FMat<'a, X> {
+    pub(crate) fn as_ptr(&'a self) -> *const ffi::fmat_t {
+        &self.fmat
+    }
+
+    pub fn length(&self) -> usize {
+        self.fmat.length as usize
+    }
+
+    pub fn height(&self) -> usize {
+        self.fmat.height as usize
+    }
+
+    pub fn print(&self) {
```

Comment:
I think this method is redundant here.

Reply:
if you actually need it you may implements Display/Debug traits.

Reply:
I've removed it, I'll implement some `Debug` trait in another PR if I ever need it :)